xrpl/models/ledger_objects/xchain_owned_create_account_claim_id.py

```diff
@@ -0,0 +1,127 @@
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import List, Optional, Union
+
+from xrpl.models.ledger_objects.ledger_entry_type import LedgerEntryType
+from xrpl.models.ledger_objects.ledger_object import LedgerObject
+from xrpl.models.nested_model import NestedModel
+from xrpl.models.required import REQUIRED
+from xrpl.models.utils import require_kwargs_on_init
+from xrpl.models.xchain_bridge import XChainBridge
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class XChainOwnedCreateAccountClaimID(LedgerObject):
+    """The model for the `XChainOwnedCreateAccountClaimID` Ledger Object"""
+
+    account: str = REQUIRED  # type: ignore
+    """
+    The account that owns this object. This field is required.
+    """
+
+    ledger_index: str = REQUIRED  # type: ignore
+    """
+    The ledger index is a hash of a unique prefix for `XChainOwnedClaimID`s, the actual
+    `XChainClaimID` value, and the fields in `XChainBridge`. This field is required.
+    """
+
+    xchain_account_create_count: Union[int, str] = REQUIRED  # type: ignore
+    """
+    An integer that determines the order that accounts created through cross-chain
+    transfers must be performed. Smaller numbers must execute before larger numbers.
+    This field is required.
+    """
+
+    xchain_bridge: XChainBridge = REQUIRED  # type: ignore
+    """
+    The door accounts and assets of the bridge this object correlates to.
+    This field is required.
+    """
+
+    xchain_create_account_attestations: List[XChainCreateAccountProofSig] = REQUIRED  # type: ignore
+    """
+    Attestations collected from the witness servers. This includes the parameters
+    needed to recreate the message that was signed, including the amount, which chain
+    (locking or issuing), optional destination, and reward account for that signature.
+    This field is required.
+    """
+
+    owner_node: str = REQUIRED  # type: ignore
+    """
+    A hint indicating which page of the sender's owner directory links to this entry,
+    in case the directory consists of multiple pages. This field is required.
+    """
+
+    previous_txn_id: str = REQUIRED  # type: ignore
+    """
+    The identifying hash of the transaction that most recently modified this object.
+    This field is required.
+    """
+
+    previous_txn_lgr_seq: int = REQUIRED  # type: ignore
+    """
+    The index of the ledger that contains the transaction that most recently modified
+    this object. This field is required.
+    """
+
+    flags: int = 0
+    """
+    A bit-map of boolean flags. Flags is always 0 since there are no flags defined for
+    XChainOwnedCreateAccountClaimID entries.
+    """
+
+    ledger_entry_type: LedgerEntryType = field(
+        default=LedgerEntryType.XCHAIN_OWNED_CREATE_ACCOUNT_CLAIM_ID,
+        init=False,
+    )
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class XChainCreateAccountProofSig(NestedModel):
```

Comment:
Is this not replicated between `XChainOwnedClaimID` and `XChainOwnedCreateAccountClaimID`?

Reply:
Not quite, they differ in class names and one has an extra field:
- `XChainCreateAccountProofSig` - has another field named `signature_reward`
- `XChainClaimProofSig`